test/Microsoft.Azure.WebJobs.Host.EndToEndTests/InvocationFilterEndToEndTests.cs

```diff
@@ -0,0 +1,460 @@
+            host.Configuration.LoggerFactory = loggerFactory;
+            host.Configuration.TypeLocator = new FakeTypeLocator(typeof(FilterOrderTestClass), typeof(FilterOrderTestClass.SampleClassForFilterOrderTest));
+            var method = typeof(FilterOrderTestClass).GetMethod("Main", BindingFlags.Public | BindingFlags.Instance);
+
+            // Invoke with method name
+
+            try
+            {
+                await host.CallAsync(method);
+            }
+            catch (Exception e)
+            {
+                var message = e.Message;
+            }
+            
+            var loggerToTest = _loggerProvider.CreatedLoggers.Where(l => l.Category == LogCategories.Executor).Single();
+            Assert.NotNull(loggerToTest.LogMessages.SingleOrDefault(p => p.FormattedMessage.Contains("A.Both")));
+            //Assert.NotNull(loggerToTest.LogMessages.SingleOrDefault(p => p.FormattedMessage.Contains("B.Both")));
+            Assert.NotNull(loggerToTest.LogMessages.SingleOrDefault(p => p.FormattedMessage.Contains("C.Both")));
+        }
+
+        [Fact]
+        public async Task TestInvocationLoggingFilter()
+        {
+            // Setup the test in memory
+            var activator = new FakeActivator();
+            var loggerFactory = new LoggerFactory();
+            var logger = new MyLogger();
+            activator.Add(new StandardFilterTests());
+            var host = TestHelpers.NewJobHost<StandardFilterTests>(activator, logger);
+            loggerFactory.AddProvider(_loggerProvider);
+            host.Configuration.LoggerFactory = loggerFactory;
+            host.Configuration.TypeLocator = new FakeTypeLocator(typeof(StandardFilterTests));
+            var method = typeof(StandardFilterTests).GetMethod("UseLoggingFilter", BindingFlags.Public | BindingFlags.Instance);
+
+            // Invoke the method
+            await host.CallAsync(method);
+
+            // Make sure the logs are correct
+            var loggerToTest = _loggerProvider.CreatedLoggers.Where(l => l.Category == LogCategories.Executor).Single();
+            Assert.NotNull(loggerToTest.LogMessages.SingleOrDefault(p => p.FormattedMessage.Contains("Test executing!")));
+            Assert.NotNull(loggerToTest.LogMessages.SingleOrDefault(p => p.FormattedMessage.Contains("Test executed!")));
+        }
+
+        [Fact]
+        public async Task TestHTTPRequestFilter()
+        {
+            // Setup the test in memory
+            var activator = new FakeActivator();
+            var loggerFactory = new LoggerFactory();
+            var logger = new MyLogger();
+            activator.Add(new StandardFilterTests());
+            var host = TestHelpers.NewJobHost<StandardFilterTests>(activator, logger);
+            loggerFactory.AddProvider(_loggerProvider);
+            host.Configuration.LoggerFactory = loggerFactory;
+            host.Configuration.TypeLocator = new FakeTypeLocator(typeof(StandardFilterTests));
```

Comment:
For some of the tests, the TypeLocator information is still required or it won't be able to find the method